ext/numo/pocketfft/pocketfftext.c

```diff
@@ -65,6 +65,9 @@ VALUE numo_pocketfft_fft(VALUE x_val, int is_forward)
     rb_funcall(z_val, rb_intern("free"), 0);
     return Qnil;
   }
+  
+  RB_GC_GUARD(x_val);
+  RB_GC_GUARD(z_val);
```

Comment:
Unfortunately, I failed to reproduce the segmentation error in my environment. I think adding only `RB_GC_GUARD (x_val);` will fix the error. Could you please try the fix?

Reply:
Works with only `x_val`.  I'll update the pull request.

----

Testing details:

I ran my scripts that reproduce the error against the current release of numo-pocketfft, to make sure the error still occurs:

```bash
lsb_release -a
# No LSB modules are available.
# Distributor ID: Ubuntu
# Description:    Ubuntu 20.04.1 LTS
# Release:        20.04
# Codename:       focal

gcc --version
# gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

i=1; while rspec; do sleep 0.1; i=$(($i+1)); done; echo "Reached $i iterations before failing"
# ...
# Aborted (core dumped)
# Reached 2 iterations before failing

j=1; while bundle exec bin/reproduce_pocket_fft_crash.rb ; do sleep 0.1; j=$(($j+1)); done; echo "Reached $j iterations before failing"
# ...
# Aborted (core dumped)
# Reached 1 iterations before failing
```

I tested again with `RB_GC_GUARD(x_val)` only.  Both tests reached 30 iterations without crashing.